### PR TITLE
Profiling

### PR DESCRIPTION
**Pull request by jhaux**
Introducing: The Pytorch Profiling hook! Saves a trace, that can be interactively viewed via the ``chrome://tracing`` tool. Take a look at the attached pictures.

Still on the todo list:

- [ ] Add warning that traces get huge fast! Solution: `--num_steps 20``
- [ ] Add example interpretation (?)

Image 1: Some 15 training steps. Note the validation steps every 2^n steps.
zoom_0
Image 2: Zoom to about one step. Left: Forward pass, middle: gradient calculation, right: gradient application.
zoom_1
Image 3: Some Conv - Batch-Norm - ReLU layers
zoom_2

### CODE REVIEW

File: edflow/hooks/logging_hooks/pt_profiling.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+from edflow.hooks.hook import Hook
+from edflow.project_manager import ProjectManager as P
+from edflow.custom_logging import get_logger
+
+
+class PyProfilingHook(Hook):
+    '''Allows to profile your pytorch code!
+
+    Best of all: It allows you to create chrome traces, which can be view
```
Comment (pesser) [inline edflow/hooks/logging_hooks/pt_profiling.py:13]: view -> viewed

**Review by pesser — APPROVED** [review]
Nice, I love it! I was always wondering if there is such a functionality which can be integrated. I guess you just need to reformat and fix the typo if you feel like doing so ;)

File: edflow/hooks/logging_hooks/pt_profiling.py (hunk 1 above)
Comment (pesser) [reply edflow/hooks/logging_hooks/pt_profiling.py:13]: btw. your docstring is great and allows one to get going immediately, but if you have a link to an exhaustive documentation of the profiler you could add it, too!

**Comment by theRealSuperMario** [comment]
This needs to be documented, otherwise it is just another awesome feature which nobody will ever know about and we will kick it out again after some time.

Besides that, I love it!

**Comment by theRealSuperMario** [comment]
black formatting is missing
